registry/registry-local/src/main/java/com/alipay/sofa/rpc/registry/local/DomainRegistry.java

```diff
@@ -0,0 +1,273 @@
+            tmpProviderInfoList.addAll(providerInfos);
+        }
+        return new ProviderGroup(RpcConstants.ADDRESS_DIRECT_GROUP, tmpProviderInfoList);
+    }
+
+    protected List<ProviderInfo> resolveDomain(String directUrl) {
+        List<ProviderInfo> providerInfos = domainCache.get(directUrl);
+        if (providerInfos != null) {
+            return providerInfos;
+        }
+        ProviderInfo providerInfo = ProviderHelper.toProviderInfo(directUrl);
+        List<ProviderInfo> result = directUrl2IpUrl(providerInfo, domainCache.get(directUrl));
+        domainCache.put(directUrl, result);
+        return result;
+    }
+
+    protected List<ProviderInfo> directUrl2IpUrl(ProviderInfo providerInfo, List<ProviderInfo> originList) {
+        List<ProviderInfo> result = new ArrayList<>();
+        try {
+            String originHost = providerInfo.getHost();
+            String originUrl = providerInfo.getOriginUrl();
+            InetAddress[] addresses = InetAddress.getAllByName(originHost);
+            if (addresses != null && addresses.length > 0) {
+                Map<ProviderInfo, ProviderInfo> originMap = originList == null ? EMPTY_MAP : originList.stream().collect(Collectors.toMap(Function.identity(), Function.identity()));
+                String firstHost = addresses[0].getHostAddress();
+                if (firstHost == null || firstHost.equals(providerInfo.getHost())) {
+                    addProviderInfo(result, originMap, providerInfo);
+                    return result;
+                } else if (StringUtils.isNotBlank(originUrl)) {
+                    for (InetAddress address : addresses) {
+                        String newHost = address.getHostAddress();
+                        ProviderInfo tmp = providerInfo.clone();
+                        String newUrl = originUrl.replace(originHost, newHost);
+                        tmp.setOriginUrl(newUrl);
+                        tmp.setHost(newHost);
+                        addProviderInfo(result, originMap, tmp);
+                    }
+                    return result;
+                }
+            }
+        } catch (Exception e) {
+            LOGGER.error("directUrl2IpUrl error", e);
+        }
+
+        List<ProviderInfo> providerInfos = new ArrayList<>();
+        providerInfos.add(providerInfo);
+        return providerInfos;
+
+    }
+
+    protected void addProviderInfo(List<ProviderInfo> target, Map<ProviderInfo, ProviderInfo> originMap, ProviderInfo providerInfo) {
+        ProviderInfo originProviderInfo = originMap.get(providerInfo);
+        if (originProviderInfo != null) {
+            target.add(originProviderInfo);
+        } else {
+            target.add(providerInfo);
+        }
+    }
+
+    @Override
+    public boolean start() {
+        return false;
+    }
+
+    @Override
+    public void register(ProviderConfig config) {
+        throw new UnsupportedOperationException("DomainRegistry not support register providerConfig:" +
```

Comment:
If the exception is expected, consider can only print the info Log here. The exeception here will cause misunderstand in monitor.

Reply:
If register method of DomainRegistry is invoked, it indicates that the program must have problems. We should fail fast in this situation.